crates/hir_ty/src/checker.rs

```diff
@@ -85,7 +85,7 @@ impl<'a> FunctionTypeChecker<'a> {
     fn check_stmt(&mut self, stmt: &'a hir::Stmt) {
         match stmt {
             hir::Stmt::ExprStmt { expr, .. } => self.check_expr(*expr),
-            hir::Stmt::VariableDef { value, .. } => self.check_expr(*value),
+            hir::Stmt::Let { value, .. } => self.check_expr(*value),
```

Comment:
**Optimization Issue**: The method check_expr is called for each value in a Let statement. This could be inefficient if the expression evaluation is complex or if it involves database access or other I/O operations. <br> **Fix**: Consider caching the results of complex expressions or using a more efficient data retrieval strategy to avoid redundant computations and I/O operations. <br> **Code Suggestion**: 
 ```
 - self.check_expr(*value),
 + self.check_expr_cached(*value),
 ```